tests/testthat/test-model-compile.R

```diff
@@ -496,15 +505,22 @@ test_that("cpp_options work with settings in make/local", {
 
   rebuild_cmdstan()
   mod <- cmdstan_model(stan_file = stan_program)
-  expect_null(mod$cpp_options()$STAN_THREADS)
+  expect_null(
+    expect_warning(mod$cpp_options()$stan_threads, "Use mod\\$exe_info()")
+  )
+  expect_false(mod$exe_info()$stan_threads)
+  expect_null(mod$precompile_cpp_options()$stan_threads)
 
   file.remove(mod$exe_file())
 
   cmdstan_make_local(cpp_options = list(stan_threads = TRUE), append = TRUE)
 
   file <- file.path(cmdstan_path(), "examples", "bernoulli", "bernoulli.stan")
   mod <- cmdstan_model(file)
-  expect_true(mod$cpp_options()$STAN_THREADS)
+  expect_true(
+    expect_warning(mod$cpp_options()$stan_threads, "Use mod\\$exe_info()")
```

Comment:
Does `expect_warning on its own not work here?

Reply:
I guess using both is testing that there's a warning and also that the value is `TRUE`?

Reply:
@jgabry exactly. It depends what you want to test. `expect_warning` checks if a warning is emitted. `expect_true` tests the return value. (Or at least that's the intention. I can double-check that `expect_warning` returns the value of the expression and not the result of the check.

Reply:
I'm pretty sure you're right that it does, so this should be OK.